### ISSUE TITLE
MultiImagePicker.pickImages() not working

### ISSUE THREAD

**Issue by agnoam** [Sh1d0w/multi_image_picker#49]
Hey I'm using this gallery library, it's awesome. But one thing to repair - I'm calling this line of code, but I didn't get any response to this array. 

`List<Asset> resultList = await MultiImagePicker.pickImages(maxImages: _maxImages - _selectedImages.length);` 

I have more one question, there's any way to send these Asset over HTTP request ?

Last thing - There is any way to enable just images pick, without any videos at the picker ?

**Comment by Sh1d0w** [comment]
Hi, please provide more info regarding the issue - what OS, output of `flutter doctor`, and also are you able to run the example app provided in this repo?

Regarding the uploading, check out the official docs: https://sh1d0w.github.io/multi_image_picker/#/upload?id=server-api

**Comment by agnoam** [comment]
This is all the logs at the time I choose 1 picture and click sure.

```
I/zygote64(30880): Do partial code cache collection, code=100KB, data=117KB
I/zygote64(30880): After code cache collection, code=100KB, data=118KB
I/zygote64(30880): Increasing code cache capacity to 512KB
I/zygote64(30880): Compiler allocated 7MB to compile void android.widget.TextView.<init>(android.content.Context, android.util.AttributeSet, int, int)
D/OpenGLRenderer(30880): endAllActiveAnimators on 0x6ffd8c7400 (AppCompatTextView) with handle 0x701ccff7c0
I/zygote64(30880): Compiler allocated 6MB to compile void android.view.ViewRootImpl.performTraversals()
```

And my device is Xiaomi A1.

**Comment by Sh1d0w** [comment]
Ok, let me know if the example app has the same issue for you. Also can you tell me what version of the plugin do you use?

**Comment by agnoam** [comment]
This is the package I'm using: `  multi_image_picker: ^2.3.27`

**Comment by agnoam** [comment]
This the example project you want to see:
```
import 'package:flutter/material.dart';
import 'package:multi_image_picker/multi_image_picker.dart';


void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() async {
//    setState(() async {

//    });
  }

  void _openGallery() async {
    String error;

    try {
      List<Asset> resultList = await MultiImagePicker.pickImages(maxImages: 20);
      resultList.forEach((result) => print(result));
    } catch (e) {
      error = e.message;
    }
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.display1,
            ),
            RaisedButton(onPressed: _openGallery,)
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

But on this project the gallery not opening, and throws this exception:
```
E/MethodChannel#multi_image_picker( 6440): Failed to handle method call
E/MethodChannel#multi_image_picker( 6440): java.lang.IllegalStateException: Reply already submitted
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.view.FlutterNativeView$1.reply(FlutterNativeView.java:174)
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:208)
E/MethodChannel#multi_image_picker( 6440): 	at com.vitanov.multiimagepicker.MultiImagePickerPlugin.finishWithError(MultiImagePickerPlugin.java:663)
E/MethodChannel#multi_image_picker( 6440): 	at com.vitanov.multiimagepicker.MultiImagePickerPlugin.finishWithAlreadyActiveError(MultiImagePickerPlugin.java:658)
E/MethodChannel#multi_image_picker( 6440): 	at com.vitanov.multiimagepicker.MultiImagePickerPlugin.onMethodCall(MultiImagePickerPlugin.java:235)
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:200)
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:163)
E/MethodChannel#multi_image_picker( 6440): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#multi_image_picker( 6440): 	at android.os.MessageQueue.next(MessageQueue.java:325)
E/MethodChannel#multi_image_picker( 6440): 	at android.os.Looper.loop(Looper.java:147)
E/MethodChannel#multi_image_picker( 6440): 	at android.app.ActivityThread.main(ActivityThread.java:6590)
E/MethodChannel#multi_image_picker( 6440): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#multi_image_picker( 6440): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
E/MethodChannel#multi_image_picker( 6440): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
E/MethodChannel#multi_image_picker( 6440): Failed to handle method call
E/MethodChannel#multi_image_picker( 6440): java.lang.IllegalStateException: Reply already submitted
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.view.FlutterNativeView$1.reply(FlutterNativeView.java:174)
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:208)
E/MethodChannel#multi_image_picker( 6440): 	at com.vitanov.multiimagepicker.MultiImagePickerPlugin.finishWithError(MultiImagePickerPlugin.java:663)
E/MethodChannel#multi_image_picker( 6440): 	at com.vitanov.multiimagepicker.MultiImagePickerPlugin.finishWithAlreadyActiveError(MultiImagePickerPlugin.java:658)
E/MethodChannel#multi_image_picker( 6440): 	at com.vitanov.multiimagepicker.MultiImagePickerPlugin.onMethodCall(MultiImagePickerPlugin.java:235)
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:200)
E/MethodChannel#multi_image_picker( 6440): 	at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:163)
E/MethodChannel#multi_image_picker( 6440): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#multi_image_picker( 6440): 	at android.os.MessageQueue.next(MessageQueue.java:325)
E/MethodChannel#multi_image_picker( 6440): 	at android.os.Looper.loop(Looper.java:147)
E/MethodChannel#multi_image_picker( 6440): 	at android.app.ActivityThread.main(ActivityThread.java:6590)
E/MethodChannel#multi_image_picker( 6440): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#multi_image_picker( 6440): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
E/MethodChannel#multi_image_picker( 6440): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

```

**Comment by agnoam** [comment]
> Hey I'm using this gallery library, it's awesome. But one thing to repair - I'm calling this line of code, but I didn't get any response to this array.
> 
> `List<Asset> resultList = await MultiImagePicker.pickImages(maxImages: _maxImages - _selectedImages.length);`
> 
> I have more one question, there's any way to send these Asset over HTTP request ?
> 
> Last thing - There is any way to enable just images pick, without any videos at the picker ?

And what about the filter I talked about ?

**Comment by Sh1d0w** [comment]
The picker already filters only images, you should not be able to see videos:

https://github.com/Sh1d0w/multi_image_picker/blob/master/android/src/main/java/com/vitanov/multiimagepicker/MultiImagePickerPlugin.java#L529

Can you provide some images where I can reproduce the problem?

**Comment by agnoam** [comment]
something like this mixed with my pictures: 

![untitled](https://user-images.githubusercontent.com/6377503/50558184-b0d7d880-0cf4-11e9-9406-9c3e87524312.png)

and I found more thing to repair: once I open the picker and exit without select any photo, I need to click twice on the button for the picker will show again

**Comment by Sh1d0w** [comment]
@agnoam Closing things seems like a bug I will fix it.

The picker does not support animated gif, in matter of fact it only supports JPEG images. I will see how I can filter those out from the picker.

**Comment by agnoam** [comment]
thanks a lot, but what with my issue with the result ?

**Comment by agnoam** [comment]
What about the result problem that I not receive any result from the picker ?

**Comment by agnoam** [comment]
Just JPEG, or any type of images like png etc. ?

**Comment by Sh1d0w** [comment]
Hi this should be fixed now in version 2.3.29.
